api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -61,6 +61,14 @@ private Types() {}
   private static final Pattern DECIMAL =
       Pattern.compile("decimal\\(\\s*(\\d+)\\s*,\\s*(\\d+)\\s*\\)");
 
+  public static Type typeFromTypeString(String typeString) {
```

Comment:
Variant isn't a primitive, but it is still better to reuse `fromPrimitiveString`. Callers are using this to parse type names, not to restrict parsing to only primitive types.

Reply:
I also think that if we want to have an additional method to avoid confusion (i.e. "I don't want to `fromPrimitiveString` that because I need support for variant") then we can add a synonym. In that case, I think there's probably a better name, like `fromString`. That's not great because it implies that it would support structs, maps, and lists, but so does `typeFromTypeString` so it's at least more direct. Unfortunately, we don't have a word in Iceberg that means a type that is expressed in a single string (vs a JSON-defined type).

Reply:
Instead of adding new  `typeFromTypeString()`, I change `fromPrimitiveString()` to `fromTypeString()` to avoid confusion. Let me know if that works for you.

Reply:
I'm thinking that `fromPrimitiveString` is a little misleading so prefer `fromTypeString`.  But let me know and I can change back.

Reply:
It's public API and renaming actually will cause incompatibility.  I renamed back to `fromPrimitiveString()` now.

Reply:
Actually it will introduces incompatibility since the return type changes. Let me know if that works or maybe we can introduce a new one as I did initially. 

> old: method org.apache.iceberg.types.Type.PrimitiveType org.apache.iceberg.types.Types::fromPrimitiveString(java.lang.String)
>   new: method org.apache.iceberg.types.Type org.apache.iceberg.types.Types::fromPrimitiveString(java.lang.String)
>

Reply:
I'd be +1 on `fromTypeString`.

Reply:
@amogh-jahagirdar  Do you prefer to add a new `fromTypeString` or change `fromPrimitiveString` to `fromTypeString` (that introduces incompatibility in the api)? Would it be fine to introduce incompatibility - assume that will be in 1.8?

Reply:
Maybe it's better just to change the method to `fromTypeString`() instead of introducing a new one to keep the api simple. Please take another look.

Reply:
I commented above, but I think this should be `fromTypeName` to make it more clear that this is not able to construct nested types. I would also keep `fromPrimitiveString` and not deprecate it because it is still useful. We just need to replace all internal uses with `fromTypeName`.